core/chains/evm/client/errors.go

```diff
@@ -153,14 +155,16 @@ var arbitrumFatal = regexp.MustCompile(`(: |^)(invalid message format|forbidden
 var arbitrum = ClientErrors{
 	// TODO: Arbitrum returns this in case of low or high nonce. Update this when Arbitrum fix it
 	// Archived ticket: story/16801/add-full-support-for-incorrect-nonce-on-arbitrum
-	NonceTooLow:           regexp.MustCompile(`(: |^)invalid transaction nonce$|(: |^)nonce too low(:|$)`),
-	NonceTooHigh:          regexp.MustCompile(`(: |^)nonce too high(:|$)`),
-	TerminallyUnderpriced: regexp.MustCompile(`(: |^)gas price too low$`),
-	InsufficientEth:       regexp.MustCompile(`(: |^)(not enough funds for gas|insufficient funds for gas \* price \+ value)`),
-	Fatal:                 arbitrumFatal,
-	L2FeeTooLow:           regexp.MustCompile(`(: |^)max fee per gas less than block base fee(:|$)`),
-	L2Full:                regexp.MustCompile(`(: |^)(queue full|sequencer pending tx pool full, please try again)(:|$)`),
-	ServiceUnavailable:    regexp.MustCompile(`(: |^)502 Bad Gateway: [\s\S]*$|network is unreachable|i/o timeout`),
+	NonceTooLow:                   regexp.MustCompile(`(: |^)invalid transaction nonce$|(: |^)nonce too low(:|$)`),
+	NonceTooHigh:                  regexp.MustCompile(`(: |^)nonce too high(:|$)`),
+	TerminallyUnderpriced:         regexp.MustCompile(`(: |^)gas price too low$`),
+	InsufficientEth:               regexp.MustCompile(`(: |^)(not enough funds for gas|insufficient funds for gas \* price \+ value)`),
+	Fatal:                         arbitrumFatal,
+	L2FeeTooLow:                   regexp.MustCompile(`(: |^)max fee per gas less than block base fee(:|$)`),
+	L2Full:                        regexp.MustCompile(`(: |^)(queue full|sequencer pending tx pool full, please try again)(:|$)`),
+	ServiceUnavailable:            regexp.MustCompile(`(: |^)502 Bad Gateway: [\s\S]*$|network is unreachable|i/o timeout`),
+	ServiceTemporarilyUnavailable: regexp.MustCompile(`(: |^)503 Service Temporarily Unavailable(:|$)`),
```

Comment:
To minimize new error cases, could we group `ServiceTemporarilyUnavailable` with the existing `ServiceUnavailable`? Seems like both are retryable so the behavior should stay the same.